File: Java/LinkedList.java
public class LinkedList{
    private Node first;
    private Node last;
    
    public boolean isEmpty(){
        return (first==null) ? true : false;
    }

    public int extract(){
        if(!isEmpty()){
            int date = first.getLoad();
            Node temp = first.getNext();
            first = temp;
            return date;
        }else{
            return -1;
        }
    }
    public int insertFirst(int date){
        Node temp = new Node (date);
        temp.setNext(first);
        first = temp;
        return date;
}
    public int insertLast(int date){
        Node temp = new Node(date);
        temp.setLoad(date);
        temp.setNext(null);
        if(first==null){
            first = temp;
            last = temp;
        }else{
            last.setNext(temp);
            last=temp;
        }
        return date;
    }
    public void clear(){
        do{
            extract();
        }while(first!=null);
        }

    public String toString(){
        String brackets = "[";
        String brackets2 = "]";
        if(isEmpty()==true)
            return brackets+"NULL"+brackets2;
        else{
            Node current = first;
            while(current.getNext() !=null){
                if(current!=last)
                    brackets = brackets + current.getLoad()+",";
                else
                    brackets = brackets + current.getLoad();
            current = current.getNext();
            }
            brackets = brackets+current.getLoad()+brackets2;
        }
        return brackets;
    }

    public boolean search(int date){
        if(isEmpty())
            return false;
        else{
            Node temp = first;
            while(temp!=null){
                if(temp.getLoad()==date)
                    return true;
                else
                    temp = temp.getNext();
            }
        }
        return false;
    }

}

File: C++/LinkedList.cpp
#include <iostream>
#include "LinkedList.h"

using namespace std;

LinkedList::LinkedList(){
	first = NULL;
	back = NULL;
}

bool LinkedList::isEmpty(){
	return (first==NULL) ? true : false; // return 0 = False 1 = True
}

int LinkedList::extract(){
	if(isEmpty()){
		return -1;
	}else{
		Node *temp = first;
		int ourValue = first->getLoad();
		first = first->getNext();
		free(temp);
		return ourValue;
	}
}
void LinkedList :: insertFirst(int n){
	Node *temp = new Node();
	temp->setLoad(n);
	temp->setNext(first);
	first = temp;
}

void LinkedList::insertLast(int n){
	Node *node = new Node();
	node->setLoad(n);
	node->setNext(NULL);
	if(first==NULL){
		first = node;
		back = node;
	}else{
		back->setNext(node);
		back = node;
	}
}

void LinkedList::toString(){
	cout<<"[";
		Node *temp = first;
		while(temp!=NULL){
			if(temp->getNext()==NULL){
				std::cout<<temp->getLoad();
			}else{
				std::cout<<temp->getLoad();
				std::cout<<",";
			}
			temp = temp->getNext();
		}
	cout<<"]";
}

bool LinkedList::search(int n){
	if(isEmpty()){
		return false;
		cout<<"Sorry, buddy, it's empty";
	}
	else{
		Node *temp = first;
		while(temp!=NULL){
			if(temp->getLoad() == n)
				return true;
			else
				temp = temp->getNext();
		}
	}
	return false;
}

void LinkedList::clear(){
	do{
		extract();
	}while(first!=NULL);
}
